frontend/model/contracts/voting/proposals.js

```diff
@@ -122,10 +123,41 @@ const proposals = {
         [RULE_DISAGREEMENT]: { threshold: 2 }
       }
     },
-    [VOTE_FOR]: function (state, { proposalHash, passPayload }) {
-      console.error('unimplemented!')
-      // TODO: unsubscribe from their mailbox and identity contract
-      //       call commit('removeContract'), etc.
+    [VOTE_FOR]: async function (state, { proposalHash, passPayload }) {
+      const proposal = state.proposals[proposalHash]
+      const { member, memberId, groupId } = proposal.data.proposalData
+      proposal.status = STATUS_PASSED
+      proposal.payload = passPayload
+
+      const data = {
+        member,
+        memberId,
+        groupId,
+        proposalHash,
+        proposalPayload: passPayload
+      }
+      const message = { data, meta: proposal.meta }
+      sbp('gi.contracts/group/removeMember/process', message, state)
+
+      removeMemberSideEffect(data)
+      // // NOTE_1: I tried to call selector sideEffect manually but it didnt work
+      // // TypeError: selectors[selector] is not a function - sbp.js
+      // sbp('gi.contracts/group/removeMember/sideEffect', {
+      //   // Q: Why sideEffect(message), message is not a simple object?
+      //   meta: () => proposal.meta,
+      //   data: () => data
+      // })
+
+      // NOTE_2: Everything went fine to all members expect the one being removed.
+      /*
+       There was an error:
+       - "processMutation: error TypeError TypeError: "state.contracts[contractID] is undefined"
+       - state.js:653 -> :172 at setContractHEAD()
+
+       I think this happens because the member received a new event where
+       the proposal was accepted. But the contract was already removed, so it breaks...
+       - A quick fix was to put a if() there but I feel like the issue is bigger than that.
```

Comment:
I think that might in fact be all that's needed to fix this, but let me think about that a bit more... Can you create an issue and assign it to me to double-check this?

Reply:
Done! #849